lib/collection/src/collection/shard_transfer.rs

```diff
@@ -257,33 +256,37 @@ impl Collection {
             .stop_task(&transfer_key)
             .await;
 
-        let replica_set =
-            if let Some(replica_set) = shard_holder_guard.get_shard(&transfer_key.shard_id) {
-                replica_set
-            } else {
-                return Err(CollectionError::bad_request(format!(
-                    "Shard {} doesn't exist",
-                    transfer_key.shard_id
-                )));
-            };
+        let mut shard_holder_guard = None;
+
+        let shard_holder = match shard_holder {
+            Some(shard_holder) => shard_holder,
+            None => shard_holder_guard.insert(self.shards_holder.read().await),
+        };
 
-        let transfer = shard_holder_guard.get_transfer(&transfer_key);
+        let Some(replica_set) = shard_holder.get_shard(&transfer_key.shard_id) else {
+            return Err(CollectionError::bad_request(format!(
+                "Shard {} doesn't exist",
+                transfer_key.shard_id
+            )));
+        };
+
+        let Some(transfer) = shard_holder.get_transfer(&transfer_key) else {
+            return Ok(());
+        };
```

Comment:
As far as I can tell, it *might* be possible for a node to propose multiple `Abort` messages (see `ShardReplicaSet::sync_local_state`, so it's a good idea to only do the following operations if transfer exists. (And as far as I can tell, it's also *safe and correct* to do so.)